modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/SortAggregateNode.java

```diff
@@ -76,7 +75,6 @@ public SortAggregateNode(
             Comparator<RowT> comp
     ) {
         super(ctx);
-        assert Objects.nonNull(comp);
```

Comment:
I would prefer to leave this assertion as is. 

Without this assertion, a situation in which we would have previously obtained either an assertion error or an NPE would now lead to the wrong result.

Perhaps, we could cover single group case by providing comparator returning 0 every time, WDYT?